taker-electron/src/lib.rs

```diff
@@ -0,0 +1,58 @@
+use neon::prelude::*;
+use once_cell::sync::OnceCell;
+use taker::Opts;
+use tokio::runtime::Runtime;
+
+// Return a global tokio runtime or create one if it doesn't exist.
+// Throws a JavaScript exception if the `Runtime` fails to create.
+fn runtime<'a, C: Context<'a>>(cx: &mut C) -> NeonResult<&'static Runtime> {
```

Comment:
Am I correct in saying that this is only called once, on line 17 in the `wrap` function? Also, does tokio not already have its own way of handling what the global runtime is? I see that this function is defined in a neon example [here](https://github.com/neon-bindings/examples/blob/855eb2d52d650afa489b6b4724095c5fdfd5c9ad/examples/tokio-fetch/src/lib.rs#L14), but I am not sure why it's done this way.

Reply:
The way I've understand it, It is the rust way of implementing a singleton pattern. Have a look 
[here](https://github.com/neon-bindings/examples/tree/855eb2d52d650afa489b6b4724095c5fdfd5c9ad/examples/tokio-fetch#executor) for more details on the boilerplate.

Reply:
That should be handled by OnceCell, shouldn't it?

https://docs.rs/tokio/1.5.0/tokio/sync/struct.OnceCell.html

Reply:
> I think this code is creating a new runtime every time, because the static is dropped as soon as we exit the runtime function.

I do not think putting the static inside of the function has any effect. This is purely a namespace/privacy change. See the example in [the Rust reference](https://doc.rust-lang.org/reference/items/static-items.html).

Reply:
> The way I've understand it, It is the rust way of implementing a singleton pattern. Have a look
[here](https://github.com/neon-bindings/examples/tree/855eb2d52d650afa489b6b4724095c5fdfd5c9ad/examples/tokio-fetch#executor) for more details on the boilerplate.

I agree that it is the standard way of making a singleton, but I don't think that one should be required here. If `start` is not called multiple times, then there is no point in making the reactor global, given that it's only used in `start` itself - it could be constructed at the beginning of the function. Correct me if I am wrong on this as I am not familiar with neon!

Reply:
@Restioson do you have a suggestion how to implement it differently?